src/azul/service/manifest_service.py

```diff
@@ -2075,9 +2078,24 @@ class ReplicaKeys:
         hub_id: str
         replica_ids: list[str]
 
-    def _replica_keys(self) -> Iterable[ReplicaKeys]:
-        request = self._create_request()
-        for hit in request.scan():
+    def _get_all_hits(self,
+                      request_callback: Callable[[Cursor | None], Search]
```

Comment:
Try to use an abstract method instead of the callback.

Reply:
Both call sites, each using a different callback, occur within the same class, so I don't see how this is possible.